### DISCUSSION TITLE
Thumbs missing for HEIC and MOV files

### DISCUSSION THREAD

**Discussion by epp137** [photoprism/photoprism#3997] [Q&A] [answered]
I messed up on my server and needed to restore from a backup which didn't include the cache folder. Now I notice that HEIC pictures and MOV files are missing thumbnails.

Initially almost all thumbs were broken but I could fix that via first rebuilding the index and then running the thumbs command. Now my log is filled with stuff like:

```
2024-01-15 01:30:19 ERRO thumb: file iPhone-FKGFWF27CX/iPhone-FKGFWF27CX/Recents/IMG_2729.HEIC.jpg is missing

2024-01-15 01:30:19 ERRO thumb: file iPhone/iPhone/Recents/IMG_2704.HEIC.jpg is missing

2024-01-15 01:23:47 ERRO thumb: file iPhone/iPhone/Recents/IMG_3108.HEIC.jpg is missing

2024-01-15 01:23:47 ERRO thumb: file iPhone/iPhone/Recents/IMG_2784.HEIC.jpg is missing

2024-01-15 01:23:47 ERRO thumb: file iPhone/iPhone/Recents/IMG_3068.HEIC.jpg is missing
```

And on the index tab it states that:

`The index currently contains 2978 hidden files. Their format may not be supported, they haven't been converted to JPEG yet or there are duplicates.`

How can I get thumbs to show? Is there maybe a command I have to run first?

**Comment by graciousgrey** [comment]
I assume that no thumbs could be created for heic files by the thumbs command in case the heic files have not been previously converted to jpg. 

Did you try a complete rescan of this folder already?

**Reply by epp137** [reply]
Yes I did that twice, should the rescan have generated jpg’s from heic files? I could run it again and check the log. Didn’t think of to do that before.

**Reply by graciousgrey** [reply]
If you are doing a [complete rescan](https://docs.photoprism.app/user-guide/library/originals/#when-should-complete-rescan-be-selected) it should create missing sidecar jpg files unless you have "Preview Images" disabled in [settings](https://docs.photoprism.app/user-guide/settings/library/#preview-images). You could also use the convert command for this.

**Reply by epp137** [reply]
I did a complete rescan, it ends with a block of messages like these:
```
2024-01-16 22:26:31 INFO indexing completed in 394 s
2024-01-16 22:26:28 INFO index: found no preview image for ps6tc8y1rhghry5b
2024-01-16 22:26:28 INFO index: found no preview image for ps6rni52n7zemij4
2024-01-16 22:26:28 INFO index: found no preview image for ps6rni5xsx2zvoe9
2024-01-16 22:26:28 INFO index: found no preview image for ps6retss5fds4gch
2024-01-16 22:26:28 INFO index: found no preview image for ps6rets16izoormv
```
I would assume for all 3922 files that are "hidden" (all of which are in HEIC format). The log is just too long to scroll to something else then those messages.
After that when I try to open one of those images the log gets an entry like
```
2024-01-16 22:29:55 ERRO thumb: file 'iPhone/iPhone/Zuletzt/IMG_4139.MOV.jpg' is missing
```
Which "convert" command do you mean @graciousgrey ? I tried "heif-convert" which created a file but it wasn't liked in photoprism. I'd need it to recognize it as a thumbnail though :/

**Reply by lastzero** [reply]
See https://docs.photoprism.app/getting-started/docker-compose/#examples

**Reply by epp137** [reply]
Thanks for the link. I ran the convert command and it successfully created jpgs. I ran a complete rescan and still I have a lot of hidden files where jpgs are missing for heics. I checked a couple and for all this is true: there is a matching jpg that has the same path and name as the heic, just in the sidecar directory. Those jpgs are not stacked with the corresponding pictures in photoprism though. So the issues seems not to be that there are no jpgs availaible but the stack somehow did not work. Any idea how that could be and what I could do to fix it?

**Reply by lastzero** [reply]
That's a lot of guesswork on our part, which is why we provide the following checklists to help with troubleshooting:
- https://docs.photoprism.app/getting-started/troubleshooting/#missing-pictures

For example, it could be that the files have been created, but the permissions are wrong, so PhotoPrism can't read them...

**Reply by lastzero** [reply]
Just to be sure: Are you using our official Docker image with the latest release (see the version in the footer of the Settings page)?
- https://docs.photoprism.app/release-notes/

**Reply by epp137** [reply]
I'm running photoprism/photoprism, version [231128-f48ff16ef](https://docs.photoprism.app/release-notes/).

I'll go through the checklists tonight and let you know what I found out.

I just made one last manual test: I deleted the jpg for a heic file in the sidecar diectory and reran the complete rescan. There was no new jpg created, I assumed the rescan would automatically do a convert if a jpg is missing. Did I assume wrong?

**Reply by lastzero** [reply]
With a complete rescan this should be the case, unless it fails for some reason, e.g. because the related sidecar folder is not writable, no suitable converter is available (e.g. if you are not using our official Docker image) or the file extension has been added to a `.ppignore` file.

**Reply by lastzero** [reply]
When you enable trace log mode, you will see more detailed logs, including the command executed and any errors it may cause (however, you will need to view them in a terminal since they are not sent to the browser for security and performance reasons):
- https://docs.photoprism.app/getting-started/troubleshooting/docker/#viewing-logs

**Reply by epp137** [reply, marked as answer]
I ran `photoprism index -f -a -c` and that fixed all my issues. Took 1.5 hours though 😁
